### PR TITLE
Remove unused includes

### PR DESCRIPTION
**Pull request by magicrub**
Removes a few includes that were unused

### CODE REVIEW

File: libraries/AP_Camera/AP_Camera.cpp [hunk 1]
```diff
@@ -9,7 +9,6 @@
 #include <GCS_MAVLink/GCS.h>
 #include <SRV_Channel/SRV_Channel.h>
 #include <AP_Logger/AP_Logger.h>
-#include <AP_GPS/AP_GPS.h>
```
Comment (WickedShell) [inline libraries/AP_Camera/AP_Camera.cpp:12]: This is wrong, GPS is needed, you are getting it transitively right now, but removing it isn't actually correct.

File: libraries/AP_Compass/AP_Compass_Calibration.cpp [hunk 2]
```diff
@@ -1,7 +1,6 @@
 #include <AP_HAL/AP_HAL.h>
 #include <AP_Notify/AP_Notify.h>
 #include <AP_GPS/AP_GPS.h>
-#include <GCS_MAVLink/GCS.h>
```
Comment (WickedShell) [inline libraries/AP_Compass/AP_Compass_Calibration.cpp:4]: We need this for doing send_text, again it's transitive right now, this doesn't actually reduce the scope of work being recompiled, and makes it a future pain point.

File: libraries/AP_Compass/CompassCalibrator.cpp [hunk 3]
```diff
@@ -62,7 +62,6 @@
 #include <AP_HAL/AP_HAL.h>
 #include <AP_Math/AP_GeodesicGrid.h>
 #include <AP_AHRS/AP_AHRS.h>
-#include <AP_GPS/AP_GPS.h>
```
Comment (WickedShell) [inline libraries/AP_Compass/CompassCalibrator.cpp:65]: Needed for fix types again...

**Review by WickedShell — CHANGES_REQUESTED** [review]
The mount changes look like the only correct parts. Everything in compass/camera is genuinely needed, and you can get away with deleting it because some other header includes it at the moment, but in general you should always include what you need locally rather then depend on some other header having brought it for you, or you increase the scope of future changes if that header manages to remove the dependency (which would be good as it would actually help build time).

**Comment by magicrub** [comment]
removed the camera and compass changes. Only the Mount ones remain

File: libraries/AP_Camera/AP_Camera.cpp (hunk 1 above)
Comment (magicrub) [reply libraries/AP_Camera/AP_Camera.cpp]: removed

File: libraries/AP_Compass/CompassCalibrator.cpp (hunk 3 above)
Comment (magicrub) [reply libraries/AP_Compass/CompassCalibrator.cpp]: removed

File: libraries/AP_Compass/AP_Compass_Calibration.cpp (hunk 2 above)
Comment (magicrub) [reply libraries/AP_Compass/AP_Compass_Calibration.cpp]: removed

**Comment by peterbarker** [comment]
Merged,thanks!

**Pull request merged**